### PR TITLE
change pom name

### PR DESCRIPTION
**Pull request by pjfanning**
use `gRPC` to match the usual style

### CODE REVIEW

**Review by mdedetrich — CHANGES_REQUESTED** [review]
I don't have an issue with this, but then we should also override the `apacheSonatypeArtifactNameProcessor` for this project so that the `gRPC` part of the name is consistent.

**Comment by pjfanning** [comment]
I understand that we don't want to complicate the sbt plugin to cover the case issue but there is no absolute requirement that this gradle-plugin pom needs to match the other poms.

File: gradle-plugin/build.gradle [hunk 1]
```diff
@@ -47,7 +47,7 @@ gradlePlugin {
 publishing {
     publications.withType(MavenPublication) {
         pom {
-            name = "Apache Pekko Grpc Gradle Plugin"
+            name = "Apache Pekko gRPC Gradle Plugin"
```
[the remark this answers is not in the document]
Comment (He-Pin) [reply gradle-plugin/build.gradle:50]: image
and the Logo is GRPC :)! maybe it's fine.  will this name cause any trademark issue?

[the remark this answers is not in the document]
Comment (He-Pin) [reply gradle-plugin/build.gradle:50]: and it was `"pekko-grpc-gradle-plugin"`.

**Comment by mdedetrich** [comment]
> I understand that we don't want to complicate the sbt plugin to cover the case issue but there is no absolute requirement that this gradle-plugin pom needs to match the other poms.

You can easily add it in this project, just do

```scala
ThisBuild / apacheSonatypeArtifactNameProcessor := { artifactId =>
  val prettified = artifactId
    .replaceAll("-", " ")
    .replaceAll("_", " ")
    .split(" ")
    .map(_.capitalize)
    .replaceAll("Grpc","gRPC")
    .mkString(" ")
  if (prettified.startsWith("Apache ")) prettified else s"Apache $prettified"
}
```

**Review by mdedetrich — APPROVED** [review]
@pjfanning I have pushed a commit that also uses this name for the rest of the sbt modules

File: build.sbt [hunk 2]
```diff
@@ -32,6 +32,11 @@ commands := commands.value.filterNot { command =>
 
 ThisBuild / reproducibleBuildsCheckResolver := Resolver.ApacheMavenStagingRepo
 
+// So that gRPC is properly styled
+ThisBuild / apacheSonatypeArtifactNameProcessor := apacheSonatypeArtifactNameProcessor.value.andThen {
+  _.replaceAll("Grpc", "gRPC")
```
Comment (He-Pin) [inline build.sbt:37]: Looks great, but better if it match something like GRPC too.

**Review by He-Pin — APPROVED** [review]
Lgtm

File: build.sbt (hunk 2 above)
Comment (pjfanning) [reply build.sbt:37]: the built plugin title cases the name - so Grpc is what is expected - making ths case insensitive is complicated and not worth the hassle

**Comment by mdedetrich** [comment]
@pjfanning You want to merge this or you still waiting on something?

**Comment by pjfanning** [comment]
merged

**Pull request merged**